### ISSUE TITLE
New checker for casting sets to lists

### ISSUE THREAD

**Issue by martimlobao** [PyCQA/pylint#4355]
### Is your feature request related to a problem? Please describe

Just had an issue in my code that took a while to debug, in particular because the output of the code was non-deterministic and there were no obvious places where randomness was being used. It turns out the issue stemmed from a place where a set was being cast as a list, which means the order of the list is not deterministic or consistent. Casting from an ordered type (set) into a type with order (list) should raise a warning. Order is an important characteristic of lists, but casting a set into a list means the order of the list is arbitrary.

### Describe the solution you'd like

A new warning (e.g. `casting-set-to-list`) should be raised when running pylint. A more general warning might be better, since casting to a tuple has the same effect (`casting-unordered-to-ordered` might be a better name). 

### Additional context

Casting a set to a list is a non-deterministic process, although users often assume it is:

```python
>>> s, t = set({1, 9}), set({9, 1})
>>> s == t
True
>>> s, t = list(s), list(t)
>>> s == t
False
```

**Comment by Pierre-Sassoulas** [comment]
This make sense. What would you recommend to fix the code ? Sorting ? If so ``s, t = sorted(list(s)), sorted(list(t))```should not raise a warning ?

**Comment by martimlobao** [comment]
One solution is to use dicts, since they maintain order since Python 3.7. There's probably some space overhead from creating a dict, but it'll continue being very performant for `.update` and similar operations:
```python
>>> l = [1, 2, 3]
>>> dict.fromkeys(l)
{1: None, 2: None, 3: None}
>>> a = dict.fromkeys(l)
>>> a.update(dict.fromkeys([3,1,0,5]))
>>> a
{1: None, 2: None, 3: None, 0: None, 5: None}
>>> list(a)
[1, 2, 3, 0, 5]
```

There are obviously cases where users want to cast to lists and don't care about order, but this should be an intentional decision done by e.g. explicitly disabling a pylint warning.
